go/chat/boxer.go

```diff
@@ -932,12 +1018,17 @@ func (b *Boxer) compareHeadersMBV2orV3(ctx context.Context, hServer chat1.Messag
 	return nil
 }
 
-func (b *Boxer) makeHeaderHash(ctx context.Context, headerSealed chat1.SignEncryptedData) (chat1.Hash, UnboxingError) {
+func (b *Boxer) makeHeaderHash(headerSealed chat1.SignEncryptedData) (chat1.Hash, UnboxingError) {
 	buf := bytes.Buffer{}
 	err := binary.Write(&buf, binary.BigEndian, int32(headerSealed.V))
 	if err != nil {
 		return nil, NewPermanentUnboxingError(err)
 	}
+	// Only the ciphertext at the end should be of variable length, otherwise
+	// this hash could be ambiguous.
+	if len(headerSealed.N) != signencrypt.NonceSize {
```

Comment:
`signencrypt.NonceSize` here vs `libkb.NaclDHNonceSize`in err msg, let's just use one

Reply:
Woops. Those are actually different values too.